repairnator/repairnator-scanner/src/main/java/fr/inria/spirals/repairnator/Launcher.java

```diff
@@ -254,97 +264,88 @@ private void run(String[] args) throws JSAPException, IOException {
+            if (this.fileMode == FileMode.SLUG) {
+                googleSpreadSheetEndProcessSerializer4Bears = new GoogleSpreadSheetEndProcessSerializer4Bears(scanner, this.googleSecretPath);
+            }
+            GoogleSpreadSheetFactory.setSpreadsheetId(GoogleSpreadSheetFactory.BEAR_SPREADSHEET_ID);
         }
-        GoogleSpreadSheetInspectorTimeSerializer googleSpreadSheetInspectorTimeSerializer = new GoogleSpreadSheetInspectorTimeSerializer(googleSecretPath);
+        GoogleSpreadSheetInspectorTimeSerializer googleSpreadSheetInspectorTimeSerializer = new GoogleSpreadSheetInspectorTimeSerializer(this.googleSecretPath);
 
         this.serializers.add(jsonSerializer);
         this.serializers.add(csvSerializer);
-        this.serializers.add(googleSpreadSheetInspectorTimeSerializer);
-
-        if (mode == LauncherMode.SLUGFORBEARS || mode == LauncherMode.BUILDFORBEARS) {
-            GoogleSpreadSheetFactory.setSpreadsheetId(GoogleSpreadSheetFactory.BEAR_SPREADSHEET_ID);
-        } else {
-            this.serializers.add(new GoogleSpreadSheetNopolSerializer(googleSecretPath));
-        }
-
         this.serializers.add(googleSpreadSheetInspectorSerializer);
-
-        Launcher.LOGGER.debug("Start to scan projects in travis...");
-
-        ProjectScanner scanner = new ProjectScanner(lookupDays);
-
-        if (mode == LauncherMode.SLUG) {
-            this.endProcessSerializer = new GoogleSpreadSheetEndProcessSerializer(scanner, googleSecretPath);
-        }
-        GoogleSpreadSheetEndProcessSerializer4Bears googleSpreadSheetEndProcessSerializer4Bears = null;
-        if (mode == LauncherMode.SLUGFORBEARS || mode == LauncherMode.BUILDFORBEARS) {
-            googleSpreadSheetEndProcessSerializer4Bears = new GoogleSpreadSheetEndProcessSerializer4Bears(scanner, googleSecretPath);
-        }
-
-        jsonSerializer.setScanner(scanner);
+        this.serializers.add(googleSpreadSheetInspectorTimeSerializer);
 
         List<BuildToBeInspected> buildsToBeInspected = null;
 
         String completeWorkspace = null;
 
-        switch (mode) {
-            case BUILD:
-                buildsToBeInspected = scanner.getListOfFailingBuildsFromGivenBuildIds(input, mode);
-                break;
-
-            case SLUG:
-                buildsToBeInspected = scanner.getListOfFailingBuildsFromProjects(input, mode);
-                break;
-
-            case SLUGFORBEARS:
-                buildsToBeInspected = scanner.getListOfPassingBuildsFromProjects(input, mode);
-                break;
-
-            case BUILDFORBEARS:
-                buildsToBeInspected = scanner.getListOfPassingBuildsFromGivenBuildIds(input, mode);
-                break;
+        if (this.launcherMode == LauncherMode.REPAIRNATOR) {
+            if (this.fileMode == FileMode.SLUG) {
+                buildsToBeInspected = scanner.getListOfFailingBuildsFromProjects(this.input, this.launcherMode);
```

Comment:
Actually, if we pass the launcherMode at construction time to the scanner, it would be able to select directly the right method based on the mode: it should be its responsibility, not the Launcher's one.